autobump-maven/src/test/java/com/github/autobump/maven/model/MavenDependencyResolverTest.java

```diff
@@ -182,11 +186,24 @@ void pluginwithProperties() {
     }
 
     @Test
-    void pluginWithNonExestentProperties() {
+    void pluginWithNonExistentProperties() {
         pluginWorkspace = new Workspace(pluginWorkspace.getProjectRoot() + "/nonExistentproperties");
         Set<Dependency> plugins = resolver.resolve(pluginWorkspace);
         assertEquals(
                 Set.of(),
                 plugins);
     }
+
+    @Test
+    void testResolveMultiModuleProject() throws Exception {
+        Set<Dependency> dependencies = resolver.resolve(multiModuleWorkspace);
+        assertEquals(3, dependencies.size());
+    }
+
+    @Test
+    void testThrowsIO() {
+        MavenDependencyResolverTester tester = new MavenDependencyResolverTester();
```

Comment:
In case we need to create a subtype specifically for testing, we often opt to define this as an anonymous inner class in the test itself. This keeps the extends close to the place where you use it. 

But this depends on personal preference of course.

```
void testThrowsIO {
    var longNameThatExplainsWhatTheExtendedTesterDoes = new MavenDependencyResolverTester() {
        @Override
        public Set<Dependency> resolveModules(Workspace workspace) throws IOException {
            throw new IOException();
        }
    }

    longNameThatExplainsWhatTheExtendedTesterDoes.resolve(multiModuleWorkspace);
}

```